com.microsoft.mrtk.input/Assets/Prefabs/MRTK Hand Controller.prefab

```diff
@@ -178,6 +179,18 @@ MonoBehaviour:
   m_Name: 
   m_EditorClassIdentifier: 
   dwellTriggerTime: 0.3
+--- !u!114 &5253881778346564948
+MonoBehaviour:
+  m_ObjectHideFlags: 0
+  m_CorrespondingSourceObject: {fileID: 0}
+  m_PrefabInstance: {fileID: 0}
+  m_PrefabAsset: {fileID: 0}
+  m_GameObject: {fileID: 33891623947274355}
+  m_Enabled: 1
+  m_EditorHideFlags: 0
+  m_Script: {fileID: 11500000, guid: 8283d2605e0565d4baff78feb5af8df7, type: 3}
```

Comment:
I see two instances of the polyfill being added (here and line 1037 below). Just wanted to make sure that was intentional, since I think this base hand controller prefab only has one pointing ray.

Reply:
Yep, one for GazePinch and one for Ray.

Can't do the polyfill on the controller because it fights the builtin "posedriver-ness" of the controller. That's part of the reason why I think this polyfill is so gross.

Reply:
Why does gaze+pinch need the polyfill? Doesn't it just depend on the gaze ray for targeting and fall back to the palm for manip pose?

> Can't do the polyfill on the controller because it fights the builtin "posedriver-ness" of the controller. That's part of the reason why I think this polyfill is so gross.

I'm also thinking about this statement a bit. Could we override `UpdateTrackingInput` in ArticulatedHandController to calculate/apply to polyfill? Like we overrode `UpdateInput`

Reply:
We can also leave the second part for a potential revision (or never, if you've already found it doesn't work!), if we'd rather get this out there sooner 😬

Reply:
Good point about it potentially being inside the controller itself.

As to "why gaze-pinch needs it", it's because gazepinch requires a pointing pose for its manipulation logic. It needs the full ray calculation becaues that's what drives the overall manipulation; it's not a 1:1 to just palm pose, it uses the ray as part of its calculation. It's essentially a far ray interaction, but with a "virtual" targeting system. 

But you're absolutely right that we could build it into the UpdateTrackingInput method. Hmmm.....

Reply:
Done and tested